api/app/GraphQL/Mutations/DuplicatePool.php

```diff
@@ -29,8 +29,17 @@ public function __invoke($_, array $args)
 
         $newPool->save();
 
-        $newPool->setEssentialPoolSkills($pool->essentialSkills->pluck('id'));
-        $newPool->setNonessentialPoolSkills($pool->nonessentialSkills->pluck('id'));
+        $skillsToSync = [];
+        foreach ($pool->poolSkills as $poolSkill) {
```

Comment:
No biggie, but this could probably be a collection map.
https://laravel.com/docs/10.x/collections#method-map

Reply:
Nice moving to the unified poolSkills attribute, though.